insights/parsers/date.py

```diff
@@ -107,6 +88,124 @@ class DateUTC(DateParser):
+                time is never updated, it relies on external facilities to maintain it.
+                If at all possible, use RTC in UTC by calling
+                'timedatectl set-local-rtc 0'.
+
+    Raises:
+        DateParseException: when the datetime in "Local time", "Universal time",
+                            "RTC time" are not in expected format.
+        ParseException: when the colon in each line except warning is not aligned.
+
+    Examples:
+        >>> ctl_info['ntp_service']
+        'active'
+        >>> ctl_info['system_clock_synchronized']
+        'yes'
+        >>> ctl_info['local_time']
+        datetime.datetime(2022, 11, 14, 23, 4, 6)
+    """
+    date_format = '%a %Y-%m-%d %H:%M:%S'
+
+    # unify the different names in rhel7 and rhel8
+    key_mapping = {
+        'ntp_synchronized': 'system_clock_synchronized'
+    }
+
+    def parse_content(self, content):
+        dict_key = None
+        warning_start = False
+        non_blank_line = None
+        for line in content:
+            if line.strip():
+                non_blank_line = line
+                break
+        if non_blank_line is None:
+            raise SkipException('No data in the output.')
+        try:
+            colon_index = non_blank_line.index(':')
+        except ValueError:
+            raise ParseException('No colon found, the line %s is not in expected format.' % line)
+        for line in content:
+            if not line.strip():
+                continue
+            if line[colon_index] == ':':
+                key = line[:colon_index].strip()
+                value = line[colon_index + 1:].strip()
+                dict_key = '_'.join(key.lower().split())
+                if dict_key in ['local_time', 'universal_time', 'rtc_time']:
+                    if dict_key == 'rtc_time':
+                        final_val = value
+                    else:
+                        final_val = value.rsplit(None, 1)[0]  # remove tz info
+                    try:
+                        self[dict_key] = datetime.strptime(final_val, self.date_format)
+                    except Exception:
+                        six.reraise(DateParseException, DateParseException(value), sys.exc_info()[2])
+                else:
+                    if dict_key in self.key_mapping:
+                        self[self.key_mapping[dict_key]] = value
+                    else:
+                        self[dict_key] = value
+            elif not line[:colon_index].strip():
+                # this line is also the content of the previous key
+                self[dict_key] += ' ' + line.strip()
+            elif line.lstrip().startswith('Warning:'):
+                warning_start = True
+                self['warning'] = line.split(':')[1].strip()
+            else:
+                if warning_start:
+                    self['warning'] += ' ' + line.strip()
+                else:
+                    raise ParseException('Unexpected format of line %s.' % line)
```

Comment:
@xiangce I found we don't need to check if dict is empty here. Since the previsous checking can make it's not empty.
Please help to review.